src/common/components/ChangeLog.tsx

```diff
@@ -0,0 +1,160 @@
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { IconProp } from '@fortawesome/fontawesome-svg-core';
+import { FC } from 'react';
+import { Button, ListGroup } from 'react-bootstrap';
+import { CircularContainer } from 'common/styles/utilities';
+import styled from 'styled-components';
+import { isKeyOfObject, isObject, isStringOrNumber } from 'common/error/utilities';
+
+export type Props = {
+  changeList: unknown[];
+  totalChanges: number;
+  changePropertyAccessor?: (change: unknown, propertyName: string) => string | number;
+  getChangeItemAction?: (action: string) => string;
+  handleShowAllChanges: () => void;
+};
+
+const StyledCircularContainer = styled(CircularContainer)`
+  background-color: lightgrey;
+  height: 32px;
+  width: 32px;
+`;
+
+const StyledEditorName = styled.span`
+  color: ${props => props.theme.changelogs.accentTextColor};
+`;
+
+const StyledShowAllChangesButton = styled(Button)`
+  color: ${props => props.theme.changelogs.accentTextColor};
+`;
+
+const defaultPropertyAccessor = (
+  change: unknown,
+  propertyName: string,
+): string | number | Record<string, unknown> | boolean => {
+  if (isObject(change)) {
+    const value = change[propertyName];
+
+    if (isStringOrNumber(value)) {
+      return value;
+    }
+
+    if (isObject(value)) {
+      return value;
+    }
+
+    if (value === null) {
+      return false;
+    }
+
+    throw Error(`The resulting value from change[${propertyName}] is not a string, number, object, or null`);
+  }
+
+  throw Error('Given change parameter is not an object');
+};
```

Comment:
I know we had discussed the library version of this before. But I think we should just type this normally, as this adds a layer of indirection that we can add in the future, but isn't necessary up front.

I suggest something like a `HistoricalRecord<T>` generic type.